File: ruby/40.rb
# project euler 40: find d1 * d10 * d100 * d1,000 * d10,000 * d100,000 * d1000,000
t = Time.new.to_f
p = 1
n = 1
i = 0
x = 0
while true
	i += 1
	
	p *= n
	if x == 7 then break else x += 1 end
end

File: python/40.py
# Project Euler 40: find d1 * d10 * d100 * d1000 * d10000 * d100000 * d1000000

ns = ''
i = 0
while len(ns) < 1000000:
	i += 1
	ns += str(i)

product = int(ns[0:1])
for i in (10, 100, 1000, 10000, 100000, 1000000):
	product *= int(ns[i-1:i])

print(product)